src/assets/content.js

```diff
@@ -5,13 +5,56 @@
 // from the contents of the active page
 chrome.runtime.onMessage.addListener( function(request, sender, sendResponse) {
   if (request.message === 'send links') {
-    sendResponse(Array.from(document.links).map( function(val) {
-      return {
-        'href': val.href,
-        'host': val.host,
-        'tagName': val.tagName,
-        'hidden': val.hidden
-      };
-    }));
+    sendResponse(findLinks());
   }
 });
+
+function findLinks() {
+  const urlRegex = /https?:\/\/[^\s]+/;
+  const all = document.querySelectorAll('*');
+  const links = [];
+  all.forEach((element) => {
```

Comment:
What do you think of splitting this functions into three smaller pieces? To make it easier for testing later. 
Something like getLinkDataFromHref() etc...

Reply:
Yeah, that makes a lot of sense